samza-core/src/main/java/org/apache/samza/coordinator/DistributedLock.java

```diff
@@ -19,24 +19,22 @@
 
 package org.apache.samza.coordinator;
 
-import java.util.concurrent.TimeUnit;
+import java.time.Duration;
 import java.util.concurrent.TimeoutException;
 
 
-public interface DistributedLockWithState {
+public interface DistributedLock {
 
   /**
-   * Try to acquire the lock, but first check if the state flag is set. If it is set, return false.
-   * If the flag is not set, and lock is acquired - return true.
+   * Try to acquire the lock
    * @param timeout Duration of lock acquiring timeout.
-   * @param unit Time Unit of the timeout defined above.
-   * @return true if lock is acquired successfully, false if state is already set.
+   * @return true if lock is acquired successfully
    * @throws TimeoutException if could not acquire the lock.
```

Comment:
nit: update comment to reflect this happens only when the wait time to acquire lock is > timeout. not, in the general case of not able to acquire the lock.